### PR TITLE
Add regexp support

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

This PR adds support for `regexp` operator

#### Related issues

- None

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: test/oracle.regexp.test.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+var should = require('should');
+var db, User;
+
+describe('regexp', function() {
+  db = getDataSource();
+
+  var User = db.define('UserRegExp', {
+    seq: {type: Number, index: true},
+    name: {type: String, index: true, sort: true},
+    email: {type: String, index: true}
```
Comment (loay) [inline test/oracle.regexp.test.js]: Missing trailing comma causing linting error

File: test/oracle.regexp.test.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      function(cb) {
+        db.automigrate('UserRegExp', cb);
+      },
+      function(cb) {
+        async.each(beatles, User.create.bind(User), cb);
+      },
+    ], done);
+  }
+
+});
```
Comment (loay) [inline test/oracle.regexp.test.js]: Block must not be padded by blank lines causing linting errors.

File: test/oracle.regexp.test.js (hunk 1 above)
Comment (raymondfeng) [inline test/oracle.regexp.test.js]: Fixed.

**Comment by raymondfeng** [comment]
@jannyHou eslint issues fixed

**Comment by raymondfeng** [comment]
ok to test

**Pull request merged**